app/helpers/compute_resources_helper.rb

```diff
@@ -19,17 +19,29 @@ def vm_power_class(s)
 
   def vm_power_action(vm, authorizer = nil)
     opts = hash_for_power_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.identity).merge(:auth_object => @compute_resource, :permission => 'power_compute_resources_vms', :authorizer => authorizer)
-    html = vm.ready? ? { :data => { :confirm =>_("Are you sure you want to power %{act} %{vm}?") % { :act => action_string(vm).downcase.strip, :vm => vm }}, :class => "btn btn-danger" } :
-                       { :class => "btn btn-info" }
+    html = if vm.ready?
```

Comment:
Extract method to power_action_message or similar if you want, ditto for the other one (and hound will shut up)